docs/DeveloperGuide.adoc

```diff
@@ -297,58 +298,162 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 |Priority |As a ... |I want to ... |So that I can...
 |`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
 
-|`* * *` |user |add a new person |
+|`* *` |clerk|use auto-complete to predict my commands|save time on verifying its existence and correctness
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`* *` |clerk|quick-fill the command box with the suggestions of Auto-Complete|so that it reduces typing of the entire command
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* *` |clerk|refer to command history|review entered commands that maybe incorrect
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *` |clerk|quick-fill the command box with history commands|inputting last few commands is easier
+
+|`* * *` |clerk|find out the upcoming appointments for given patients. |
+
+|`* * *` |clerk|update the doctors' details by typing commands and user details. |
+
+|`* * *` |clerk|add new doctors into system |
+
+|`* *` |clerk|remove a patient from the queue if they leave. |
+
+|`* *` |clerk|view the number of patients who visited the clinic today |
+
+|`* * *` |clerk|edit patients' details | keep their particulars up to date
+
+|`* * *` |clerk|register new patients with optional fields |
+
+|`* *` |clerk|find all patients who have missed their appointments. | keep track of the list of patients whom I need to inform
+
+|`* * *` |clerk|add ad-hoc patients to the queue |
+
+|`* *` |clerk|acknowledge appointments if patients are present for their appointments | keep track of patients who came for their appointments
+
+|`* *` |clerk|tag patient with known allergies. | keep track of their allegies
+
+|`* *` |clerk|cancel appointments for patients | free up appointment time slots
+
+|`* * *` |clerk|search for patients using their name or phone number |
+
+|`* * *` |clerk|look up how many patients are in the queue, on a side panel | recommend estimated time that the patient will be attended to
+
+|`* * *` |clerk|look up patient using a reference id. |
+
+|`* * *` |clerk|reschedule appointments of patients |
+
+|`* * *` |clerk|search for appointment slots easily | schedule appointments for patients easily.
+
+|`* * *` |clerk|assign a queue number to each patient in the queue |
+
+|`* *` |clerk|schedule patient's follow up appointments |
+
+|`* * *` |clerk|use the appointment scheduler | schedule appointments for my patients
+
+|`* * *` |clerk|save time managing the queue | have more time to do my own work
+
+|`* * *` |clerk|take note of the doctors that are on-shift | effectively direct patients to available doctors.
+
+|`* *` |experienced user|create custom aliases to commands |
+
+|`* *` |receptionist|use the undo and redo feature to remedy any mistakes.|
```

Comment:
it would be good if you define the difference/relationship between 'receptionist' and 'clerk'. if 'receptionist' is a more specialised version of a 'clerk', an actor generalisation could be defined